ayoung/src/App.jsx

```diff
@@ -0,0 +1,110 @@
+import { useState, useEffect } from "react";
+import SelectPlayer from "@/components/SelectPlayer";
+import Board from "@/components/Board";
+import GameStatus from "@/components/GameStatus";
+import { calcWinner, getComputerPosition } from "@/utils/gameUtils";
+import styled from "styled-components";
+
+const App = () => {
+  const [playerMark, setPlayerMark] = useState("X");
+  const [mark, setMark] = useState("X");
+  const [squares, setSquares] = useState(Array(9).fill(null));
+  const [turnCount, setTurnCount] = useState(0);
+
+  const isPlayerTurn = playerMark === mark;
+  const winner = calcWinner(squares);
+  const isGameStarted = 0 < turnCount;
+  const isGameEnded = winner.winner || turnCount >= 9;
+
+  const handlePlayerSelect = (selectedMark) => {
+    if (playerMark === selectedMark || isGameStarted || isGameEnded) {
+      return;
+    }
+    setPlayerMark(selectedMark);
+  };
+
+  const handleClick = (idx) => {
+    if (!isPlayerTurn || isGameEnded) return;
+    updateSquares(idx, mark);
+  };
+
+  const updateSquares = (idx, currentMark) => {
+    if (squares[idx]) {
+      return;
+    }
+    const newSquares = [...squares];
+    newSquares[idx] = currentMark;
+    setTurnCount((prev) => prev + 1);
+    setSquares(newSquares);
+    setMark((prevMark) => (prevMark === "X" ? "O" : "X"));
+  };
+
+  useEffect(() => {
+    if (isPlayerTurn || isGameEnded) return;
+    const position = getComputerPosition(squares, mark);
+    if (position === null) {
+      return;
+    }
+
+    const timer = setTimeout(() => {
+      updateSquares(position, mark);
+    }, 700);
+    return () => clearTimeout(timer);
+  }, [isPlayerTurn, mark, squares]);
+
+  const handleReset = () => {
+    setSquares(Array(9).fill(null)); // 게임 보드를 초기화
+    setMark("X"); // 선공을 X로 초기화
+    setPlayerMark("X"); // 플레이어 선택을 X로 초기화
+    setTurnCount(0); // 플레이 턴 카운트 초기화
+  };
+
+  return (
+    <Container>
+      <GameStatus
+        winner={winner.winner}
+        turnCount={turnCount}
+        playerMark={playerMark}
+        isPlayerTurn={isPlayerTurn}
+      ></GameStatus>
+      <SelectPlayer
+        player={playerMark}
+        onPlayerSelect={handlePlayerSelect}
+        turnMark={isGameEnded ? null : mark}
```

Comment:
```suggestion
turnMark={!isGameEnded && mark}
```
이렇게 줄여서 사용할 수 있습니다.
`isGameEnded` 가 false인 경우에는 mark 값이 나올것이고
`isGameEnded` 가 true 인 경우는 첫번째 조건에서 걸려서 전체 표현식이 false를 노출할거예요